gnpy/core/utils.py

```diff
@@ -199,3 +199,48 @@ def rrc(ffs, baud_rate, alpha):
     p_inds = np.where(np.logical_and(np.abs(ffs) > 0, np.abs(ffs) < l_lim))
     hf[p_inds] = 1
     return sqrt(hf)
+
+def merge_dicts_recursively(dict1, dict2):
+    """Merge contents of dicts recursively
+
+    >>> d1 = {'params': {'restrictions': {'preamp_variety_list': [], 'booster_variety_list': []}}}
+    >>> d2 = {'params': {'target_pch_out_db': -20}}
+    >>> merge_dicts_recursively(d1, d2)
+    {'params': {'restrictions': {'preamp_variety_list': [], 'booster_variety_list': []}, 'target_pch_out_db': -20}}
+
+    >>> d3 = {'params': {'restrictions': {'preamp_variety_list': ['foo'], 'booster_variety_list': ['bar']}}}
+    >>> merge_dicts_recursively(d1, d3)
+    {'params': {'restrictions': {'preamp_variety_list': [], 'booster_variety_list': []}}}
+    """
+
+    copy_dict1 = dict1.copy()
+    for key in dict2:
+        if key in dict1:
+            if isinstance(dict1[key], dict):
+                copy_dict1[key] = merge_dicts_recursively(copy_dict1[key], dict2[key])
+        else:
+            copy_dict1[key] = dict2[key]
+    return copy_dict1
+
+def silent_remove(this_list, elem):
+    """Remove matching elements from a list without raising ValueError
+
+    >>> li = [0, 1]
+    >>> silent_remove(li, 1)
+    >>> li
+    [0]
+    >>> silent_remove(li, 1)
+    >>> li
+    [0]
+    """
+
+    try:
+        this_list.remove(elem)
+    except ValueError:
+        pass
+
+class EquipmentError(Exception):
+    """Raised when the json equipment is not formed correctly
+    """
+    import sys
+    sys.tracebacklimit = 0
```

Comment:
I do not think that this is a nice coding pattern. Raising an exception should not result in a random callback into the `sys` module and trying to reconfigure backtrace behavior. Also, [this variable is no longer available in Python 3](https://bugs.python.org/issue12276).

If the intention is to suppress detailed error messages, then I suggest moving any UI-affecting bits into the CLI driver, i.e., `examples/transmission_main_example.py`. I'll create another pull request for these exceptions.